src/main/java/core/basesyntax/Rectangle.java

```diff
@@ -27,4 +27,30 @@ public Integer getLength() {
     public String getColor() {
         return color;
     }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        Rectangle rectangle = (Rectangle) o;
+        if (width != null ? !width.equals(rectangle.width) : rectangle.width != null) {
+            return false;
+        }
+        if (length != null ? !length.equals(rectangle.length) : rectangle.length != null) {
+            return false;
+        }
+        return color != null ? color.equals(rectangle.color) : rectangle.color == null;
```

Comment:
nice!

let's see our checklist and try refactoring this: `While comparing fields at equals, use single return statement for all fields(e.g. return check1 && check2 && check3;).`